kernel/sched/core.c

```diff
@@ -3632,6 +3632,94 @@ int default_wake_function(wait_queue_t *curr, unsigned mode, int wake_flags,
 }
 EXPORT_SYMBOL(default_wake_function);
 
+void __setprio_other(struct rq *rq, struct task_struct *p)
+{
+	int oldprio, queued, running;
+	const struct sched_class *prev_class;
+
+	lockdep_assert_held(&rq->lock);
+
+	oldprio = p->prio;
+	prev_class = p->sched_class;
+	queued = task_on_rq_queued(p);
+	running = task_current(rq, p);
+	BUG_ON(!rt_throttled(p));
+
+	if (queued)
+		dequeue_task(rq, p, DEQUEUE_SAVE | DEQUEUE_MOVE);
+/*
+	if (running)
+		put_prev_task(rq, p);
+*/
```

Comment:
OK. Also, considering that put_prev_task_rt() would only enqueue the task in the pushable list (and we don't want that). It seems save to remove it.